pyproject.toml

```diff
@@ -32,13 +33,21 @@ reportUnnecessaryTypeIgnoreComment = true
 [tool.ruff]
 select = ["ALL"]
 ignore = [
-    # Star imports
-    "F403",
-    "F405",
-    # Pyright error codes are obnoxiously long. Ignore lint telling you to use them.
-    "PGH003",
+    "ANN002", "ANN003", "ANN101", "ANN102", "ANN201", "ANN204", "ANN206", "ANN401",
```

Comment:
Could you add comments here as to why these have all been disabled globally? `noqa` can be used in specific instances, `# ruff: noqa: ...` can be used at the top of files for file scopes, and `per-file-ignores` for directories.

Things like `C408` shouldn't be disabled and make sense:

`N999`: module names should be snake case

I commented out all of these locally apart from `ANN`, `D` and `N999` and received no lints when running `pre-commit run --all-files`.

As for all the `D` rules, `tool.ruff.pydocstyle.convention` should be changed, or the docstrings should be in numpy format. Personally I prefer the numpy syntax as it is more human readable outside of a parser, but if you disagree we can discuss. Docstrings should also be in imperative form, as specified by [PEP 257](https://peps.python.org/pep-0257/#one-line-docstrings).

As for the `ANN` rules, only the self (and cls can be disabled too) rule piped up, which I agree should be disabled. The rest should not and defeats the point of having a linter somewhat.

Reply:
Let's review which linter rules make sense for us and then going forward stick with them. We should not make our lives overly complicated, but quality and consistency of our codebase are key scoring criteria. We need to strike a good balance.

@ooliver1 you seem very familiar with the linter rules, so I suggest that you take ownership of this part.

For guidance on the documentation standard, please review the [code documentation](https://github.com/thijsfranck/the-dynamic-typists#code-documentation) section of our README. I wrote this with the numpy standard in mind, as I saw this is what we had set up in the linter. If we decide to switch, please make sure to also update the README.

Reply:
@ooliver1: I didn't really work much with ruff, and I simply ignored the things I didn't understand or thought are weird errors/warnings.
I reverted the ignores, but also removed the ones for the star import. I think star imports are horrible.